src/components/Map.svelte

```diff
@@ -29,17 +29,19 @@
   };
 </script>
 
-{#key `${map.id}-${map.index}`}
-  <div class="map">
-    <svelte:component
-      this={MapComponent}
-      index={map.index}
-      id={`${map.id}-${map.index}`}
-      mapStyle={map}
-      {numberOfMaps}
-      {...$$restProps}
-      on:mapMove
-    />
+<div class="map">
+  <svelte:component
+    this={MapComponent}
+    index={map.index}
+    id={`${map.id}-${map.index}`}
+    mapStyle={map}
+    {numberOfMaps}
+    {...$$restProps}
+    on:mapMove
+  />
+  <!-- Use the number of maps and index to reset map on adding and removing maps -->
+  <!-- We don't want to use the map id here or we'll unnecessarily remount the component for every new style -->
+  {#key `${numberOfMaps}-${map.index}`}
```

Comment:
I switched the key here to be number of maps and index so that we don't remount when changing a style while keeping the same number of maps. That remount was causing unnecessary calls to polling that were messing up the existing logic.

The unique key here originally resolved an issue where on changing the number of maps by removing them `MapStyleInput` gets confused and shows the wrong dropdown option.

I also moved the key to only wrap the map label (and therefore the map style input) since that was the original issue. Wrapping the entire map in this key will trigger total map remounts for all maps when changing the number of maps which isn't ideal. In those scenarios it's better to just resize the existing maps and only mount new maps.